benchexec/tools/smack.py

```diff
@@ -75,6 +75,12 @@ def determine_result(self, returncode, returnsignal, output, isTimeout):
             return result.RESULT_TRUE_PROP
         elif re.search(r'SMACK found an error.*', splitout):
             return result.RESULT_FALSE_REACH
+        elif re.search(r'SMACK found an invalid pointer dereference', splitout):
+            return result.RESULT_FALSE_DEREF
+        elif re.search(r'SMACK found an invalid memory deallocation', splitout):
+            return result.RESULT_FALSE_FREE
+        elif re.search(r'SMACK found a memory leak', splitout):
+            return result.RESULT_FALSE_MEMTRACK
```

Comment:
Btw., you can use `'text' in splitout` to search for a substring in Python without involving a regexp engine (should be more efficient).